### PR TITLE
Add new era in Japan called "Reiwa"

### PR DESCRIPTION
**Pull request by acid-chicken**
> 【お知らせ】新元号は「令和」に決まりました。
> &mdash; [首相官邸 on Twitter](https://twitter.com/kantei/status/1112545576472571904)

### CODE REVIEW

**Comment by krwq** [comment]
@acid-chicken we read Eras from the registry but updating fallback data is fine too I believe.

cc: @tarekgh

**Comment by tarekgh** [comment]
@acid-chicken I don't think we need to update this hard coded table as it is not really used. On Windows we read the Windows registry which is going to be updated by Windows Updates. On Linux, we use ICU which also is going to release a new version with the updated era names. 

The hard coded table is used only if we fail getting the needed information from the OS. 

@krwq I am not seeing any harm to to take this change even if it is not used. at least the information would reflect the current state. we just want to ensure this change is not going to cause any side effect. I'll double check that quickly before we decide to merge it.

**Comment by tarekgh** [comment]
@acid-chicken could you please update the table in the following places too?

https://github.com/dotnet/coreclr/blob/ff43a803a814eaaa5eba02cafa4a91def3e4c7be/src/System.Private.CoreLib/shared/System/Globalization/JapaneseCalendar.Win32.cs#L203
https://github.com/dotnet/coreclr/blob/ff43a803a814eaaa5eba02cafa4a91def3e4c7be/src/System.Private.CoreLib/shared/System/Globalization/JapaneseCalendar.WinRT.cs#L50

**Comment by acid-chicken** [comment]
@krwq @tarekgh Thanks for let me know. I just updated those tables too.

**Comment by Gnbrkm41** [comment]
It appears that some call sites of GetEraInfo() relies on the fact that the first item in the array is the current era. (e.g. `GregorianCalendarHelper _helper` field in `JapaneseCalendar`/`JapaneseLunisolarCalendar`, which then is called in various public APIs). This means that some methods may behave weirdly as it will return the value for the new era (Reiwa) instead of the current era (Heisei). It wouldn't matter if this is going to be released after the 1st of May (which then becomes perfectly valid), however until then this would lead to some unexpected behaviour.

Also, `s_japaneseErasEnglishNames` field and `GetJapaneseEnglishEraName()` method [does not appear to have been referenced in any places](https://source.dot.net/#System.Private.CoreLib/shared/System/Globalization/JapaneseCalendar.Win32.cs,d96aff29eed00499,references) (i.e. dead code); Could we just remove them after all?

File: src/System.Private.CoreLib/shared/System/Globalization/JapaneseCalendar.cs [hunk 1]
```diff
@@ -14,11 +14,11 @@ namespace System.Globalization
     /// and the era begins on the day an emperor ascends the throne and continues until his death.
```
Comment (Gnbrkm41) [inline src/System.Private.CoreLib/shared/System/Globalization/JapaneseCalendar.cs]: Maybe we should change this comment as well, he's still going strong 😀

**Comment by tarekgh** [comment]
@Gnbrkm41 the change in PR is used only when we fallback when something catastrophic happen. So, your concerns regarding having this update before May 1st are not real and no one will experience this change .  

for s_japaneseErasEnglishNames, we still need that. 

For the comment, feel free to open a PR with the changes in that comment.

**Pull request merged**